src/components/myPage/assignesAbout/my_page_main/subject_card.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { Subject } from "@/components/myPage/assignesAbout/subjects_data";
+import {TEXT_STYLES} from '@/styles/constants/textStyles';
+import { useState } from "react";
+import * as styles from "./subject_card.styles";
+import { COLORS } from "@/styles/constants/colors";
+import Link from "next/link";
+import { title } from "process";
+
+
+interface FormData {
+  title?: string;
+  semester?: string;
+  studentId?:number;
+  color:string;
+  width?:number;
+}
+
+export default function SubjectCard(f:FormData) {
+  const [formData, setFormData] = useState<FormData>({
+    title: f.title,
+    semester: f.semester,
+    studentId: f.studentId,
+    color: f.color,
+    width:f.width
+  });
+  return (
+    
+    <Link style={{display: "table-cell", verticalAlign: "middle"}}
+      href={{
+        pathname: `/my_page/my_page_my_assign`,
+        query: { 
+          subject_j: JSON.stringify(new Subject(formData.title,formData.semester, formData.studentId)),
+          color: formData.color,
+        }
+      }}
+      
+        as={`/my_page/my_page_my_assign/${formData.title}`}>
+      <styles.SubjectCard color={formData.color} width={formData.width}
+      onClick={()=>{console.log("Link Clicked");
```

Comment:
onClick이 있다면 간단한게 아니라면 의미있는 이름이 있는 함수로 분리하는게 좋아보입니다
```suggestion
      onClick={onClickCard()}
```